service/src/main/kotlin/fi/espoo/evaka/application/ApplicationStateService.kt

```diff
@@ -940,12 +936,17 @@ class ApplicationStateService(
         setCheckedByAdminToDefault(original.id, updatedForm)
         when (manuallySetDueDate) {
             null ->
-                calculateAndUpdateDueDate(
-                    today,
-                    original,
-                    updatedForm.preferences.preferredStartDate,
-                    updatedForm.preferences.urgent
-                )
+                // We don't want to calculate the due date for applications in the CREATED state.
+                // Whether this is a transfer application affects the calculation of the due date,
+                // but it's only determined when the application is sent.
+                if (original.status == SENT) {
```

Comment:
Should this be `!= CREATED`? What if service worker updates it e.g. in WAITING_FOR_PLACEMENT status, should the due date be updated as with old logic?

Reply:
There's a check on line 924 that the status is one of CREATED, SENT.

Reply:
ah 👍